akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick.internal
+
+import akka.Done
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.projection.Projection
+import akka.projection.slick.OffsetStore
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.stream.{ KillSwitch, KillSwitches }
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Future
+@InternalApi
+private[projection] class SlickProjectionImpl[Offset, StreamElement, P <: JdbcProfile](
+    projectionId: String,
+    sourceProvider: Option[Offset] => Source[StreamElement, _],
+    offsetExtractor: StreamElement => Offset,
+    databaseConfig: DatabaseConfig[P],
+    eventHandler: StreamElement => DBIO[Done])
+    extends Projection {
+
+  private val offsetStore = new OffsetStore(databaseConfig.db, databaseConfig.profile)
+
+  private var shutdown: Option[KillSwitch] = None
+
+  override def start()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
```

Comment:
I think we should guard against multiple calls to `start()`

```
private val started = new AtomicBoolean(false)

override def start()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
  if (started.compareAndSet(false, true)) {
    ...

Reply:
:+1: 

Only allow the projection to start once, but let `start` be called any time (i.e. by the projection runner).

Reply:
On the testkit PR, I have introduced a `RunningProjection`, but changed my mind. 

So, instead of managing an `AtomicBoolean`, the `start` method could return a `RunningProjection`. 

If you call it twice, you have two `RunningProjection` instances. Of course, that's illegal, but users aren't supposed to run projections by themselves. 

Then `RunningProjection` has the `stop` method and can only me stopped. 

When we move this to inside an Actor and distribute it in the cluster, the `RunningProjection` can be the state of the actor. So, when on that state (ie: `Option[RunningProjection]`), if can only stop it, nothing else. 

When `None`, if can only start a new one. 

(I'm using `Option` here just as a tool to think, in the actor we can just switch behavior)